src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -425,78 +425,109 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
 
   const typet &type=f.type();
   symbol_exprt i=fresh_symbol("parsed_int", type);
+  symbol_exprt sign=fresh_symbol("sign_of_parsed_int", type);
   const refined_string_typet &ref_type=to_refined_string_type(str.type());
   const typet &char_type=ref_type.get_char_type();
   exprt minus_char=constant_char('-', char_type);
   exprt plus_char=constant_char('+', char_type);
   assert(type.id()==ID_signedbv);
 
   exprt chr=str[0];
+  const typet &index_type=ref_type.get_index_type();
+  exprt zero_expr=from_integer(0, index_type);
+  exprt one_expr=from_integer(1, index_type);
+  exprt two_expr=from_integer(2, index_type);
```

Comment:
I went a little crazy and made almost all the variables in the function const...